{{ cookiecutter.__package_name_kebab_case }}/Dockerfile

```diff
@@ -124,6 +137,9 @@ RUN ln -s /run/secrets/poetry-auth /home/user/.config/pypoetry/auth.toml
 
 FROM base AS app
```

Comment:
I'm a little concerned about this: if we do this we'll miss out on `build-essential`, which comes with compilers but also shared libraries. If the compilers were used to compile certain packages, won't the `app` stage break without the shared libraries that those compiled packages depend on?

Reply:
Yes, that was my concern at first as well, but it simply worked even with things like NumPy installed which depend on the C shared libraries. I think the reason everything works is the following. [`build-essential` is a Debian meta-package](https://packages.ubuntu.com/focal/build-essential) which apart from libc installs a C and C++ compiler, `make` and the Debian package development tools. The only thing you actually need in the app image is libc - the compilers etc are not needed at runtime. Looking a little closer [this stage](https://hub.docker.com/layers/library/python/3.9.18-slim-bullseye/images/sha256-ac85017938f7663a8ce2e1679b8512176c15622fa70d37a33043e3333168d787?context=explore) in the official Python Docker image, we can see that libc is installed in the official Python Docker images. So at least for build-essential, there is no risk of missing shared libraries or the like. If there are other shared libraries we're compiling against those should of course be installed in the `base` layer but that's a specific use case.

Reply:
Excellent, thank you for the analysis!